harness/tests/determined/common/experimental/test_trial.py

```diff
@@ -0,0 +1,70 @@
+import datetime
+from typing import Callable
+
+import pytest
+import responses
+
+from determined.common import api
+from determined.common.experimental import trial
+from tests.fixtures import api_responses
+
+_MASTER = "http://localhost:8080"
+
+
+@pytest.fixture
+def standard_session() -> api.Session:
+    return api.Session(master=_MASTER, user=None, auth=None, cert=None)
+
+
+@pytest.fixture
+def make_trialref(standard_session: api.Session) -> Callable[[int], trial.Trial]:
+    def _make_trialref(trial_id: int) -> trial.Trial:
+        return trial.Trial(trial_id, standard_session)
+
+    return _make_trialref
+
+
+@responses.activate
+def test_trial_logs_converts_epoch_time(make_trialref: Callable[[int], trial.Trial]) -> None:
+    trialref = make_trialref(1)
+    trial_logs = api_responses.sample_get_trial_logs()
+    trial_logs_url = f"{_MASTER}/api/v1/trials/{trialref.id}/logs"
+
+    before_ts = 1689285229
+    after_ts = 1689280229
+
+    timestamp_before = datetime.datetime.fromtimestamp(before_ts).isoformat("T") + "Z"
+    timestamp_after = datetime.datetime.fromtimestamp(after_ts).isoformat("T") + "Z"
+    params = {
+        "timestampBefore": timestamp_before,
+        "timestampAfter": timestamp_after,
+    }
+    responses.get(
+        trial_logs_url,
+        stream=True,
+        body=trial_logs,
+        match=[responses.matchers.query_param_matcher(params, strict_match=False)],
+    )
+
+    resp = trialref.logs(timestamp_before=before_ts, timestamp_after=after_ts)
+
+    assert len(list(resp)) > 1
+    assert len(responses.calls) == 1
+
+    call = responses.calls[0]
+    assert call.request.params["timestampBefore"] == params["timestampBefore"]
+    assert call.request.params["timestampAfter"] == params["timestampAfter"]
+
+
+@pytest.mark.parametrize(
+    "timestamp",
+    ["2021-10-26T23:17:12+04:00", "2021-10-26 23:17:12Z", "2021-10-26T23:17:12"],
```

Comment:
There's something about this test that doesn't feel right but that I can't quite put my finger on. I think it has something to do with it being difficult to understand why, at a glance, the test is failing if I'm on call and the test fails.

One possibility is that I don't know how to differentiate between the different parameters. If that's why I'm vaguely uncomfortable, then giving them more annotation with either [the ids argument in parameterize](https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest.Metafunc.parametrize) or by [using the parameter set constructor](https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest-param) would help.

Maybe it's that I don't know exactly what each parameter is supposed to exercise? And, I dunno, maybe that's the same thing I just said? I don't understand why this is hard for me to figure out and write about.

Reply:
i think i understand your concerns. to anyone reading/debugging, it just looks like a bunch of random timestamps. i added a "name" parameter so now the pytest output looks like:
```
harness/tests/determined/common/experimental/test_trial.py::test_trial_logs_invalid_timestamps[invalid_timezone-2021-10-26T23:17:12+04:00] PASSED [ 50%]
harness/tests/determined/common/experimental/test_trial.py::test_trial_logs_invalid_timestamps[missing_T-2021-10-26 23:17:12Z] PASSED [ 75%]
harness/tests/determined/common/experimental/test_trial.py::test_trial_logs_invalid_timestamps[missing_Z-2021-10-26T23:17:12] PASSED [100%]
```
does this reduce your discomfort? 😄

Reply:
Actually... yes. Much more than I'd expected. Thanks!